### PR TITLE
[869] Fix overlap in Red Line slow zone chart

### PR DESCRIPTION
**Pull request by akh4rf**
## Motivation

Addresses #869 

## Changes

image

### CODE REVIEW

File: common/components/maps/diagrams/lines.ts [hunk 1]
```diff
@@ -76,8 +77,10 @@ export const createDefaultDiagramForLine = (
   lineName: DiagrammableLineName,
   options: CreateDiagramOptions = {}
 ) => {
-  if (lineName === 'Red') {
-    return createRedLineDiagram(options);
+  switch (lineName) {
+    case 'Red':
+      return createRedLineDiagram(options);
+    default:
+      return createStraightLineDiagram(lineName, options);
```
Comment (akh4rf) [inline common/components/maps/diagrams/lines.ts:84]: I mainly just changed this in anticipation of the green line map, which I hope to work on once we extract the map stuff into a library. It's not necessary but idk to me this is easier to read.

File: common/components/maps/diagrams/lines.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import type { Turtle } from './types';
 type DiagrammableLineName = 'Red' | 'Orange' | 'Blue';
 
 type CreateDiagramOptions = {
+  /** Number of pixels between each station */
```
Comment (devinmatte) [inline common/components/maps/diagrams/lines.ts:11]: Thank you for this! This code is sometime hard to follow given it's complexity so docs like this are very helpful

**Review by devinmatte — APPROVED** [review]
LGTM but maybe @idreyn should take a quick pass to confirm?

File: common/components/maps/diagrams/lines.ts (hunk 2 above)
Comment (akh4rf) [reply common/components/maps/diagrams/lines.ts:11]: No problem! I'm definitely going to add more of this but I figured I didn't want to bloat this PR with too many comments.
